### PR TITLE
FIX: Issues with incorrect unread and private message topic tracking state

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit fixes two issues at play. The first was introduced
in f6c852bf8e7f4dea519425ba87a114f22f52a8f4 (or maybe not introduced
but rather revealed). When a user posted a new message in a topic,
they received the `unread` topic tracking state MessageBus message,
and the Unread (X) indicator was incremented by one, because with the
aforementioned perf commit we "guess" the correct last read post
for the user, because we no longer calculate individual users' read
status there. This meant that every time a user posted in a topic
they tracked, the unread indicator was incremented. To get around
this, we can just exclude the user who created the post from the
target users of the `unread` state message.

The second issue was related to the private message topic tracking
state, and was somewhat similar. Whenever a user created a new private
message, the New (X) indicator was incremented, and could not be
cleared until the page was refreshed. To solve this, we just don't
update the topic state for the user when the `new_topic` tracking state
message comes through if the user who created the topic is the
same as the current user.

cf. https://meta.discourse.org/t/bottom-of-topic-shows-there-is-1-unread-remaining-when-there-are-actually-0-unread-topics-remaining/220817

### CODE REVIEW

**Review by pmusaraj — APPROVED** [review]
Looks good, just a few minor comments.

File: app/assets/javascripts/discourse/app/models/private-message-topic-tracking-state.js [hunk 1]
```diff
@@ -159,13 +163,14 @@ const PrivateMessageTopicTrackingState = EmberObject.extend({
...
     switch (message.message_type) {
       case "new_topic":
-        this._modifyState(message.topic_id, message.payload);
-
-        if (
-          [NEW_FILTER, INBOX_FILTER].includes(this.filter) &&
-          this._shouldDisplayMessageForInbox(message)
-        ) {
-          this._notifyIncoming(message.topic_id);
+        if (message.payload.created_by_user_id !== this.currentUser.id) {
```
[the remark this answers is not in the document]
Comment (tgxworld) [reply app/assets/javascripts/discourse/app/models/private-message-topic-tracking-state.js:166]: Ah I guess this is to handle group messages?

**Comment by martin-brennan** [comment]
@tgxworld ah you are right about the prefix, not sure why I called it DEV :laughing:

File: app/assets/javascripts/discourse/app/models/private-message-topic-tracking-state.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/models/private-message-topic-tracking-state.js:166]: Yes it's to handle group messages, it's unavoidable in this case.

File: app/models/private_message_topic_tracking_state.rb [hunk 2]
```diff
@@ -120,6 +120,10 @@ def self.publish_unread(post)
         .where("gu.group_id IN (?)", group_ids)
     end
 
+    # Note: At some point we may want to make the same peformance optimisation
```
[the remark this answers is not in the document]
Comment (martin-brennan) [reply app/models/private_message_topic_tracking_state.rb:123]: Good idea, will do!

**Pull request merged**